### PR TITLE
Add support for the Intune compliance status values for device posture checks

### PR DESCRIPTION
**Pull request by da-cf**
Currently, device posture rules support the `compliance_status` `input` field in Terraform for the Workspace One posture provider. 

Intune, a different posture provider, also uses the `compliance_status` `input` key to configure a posture rule, but we do not support the majority of configuration values for it. 

Technically, the `compliance_status` key can be used for an `intune` device posture rule `type` if the value is `compliant` or `noncompliant`. However, this is not documented for the `compliance_status` field, which states it is for workspace one only ([source](https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/internal/sdkv2provider/schema_cloudflare_device_posture_rule.go#L150)).

Workspace One AND Intune both use the `compliant` and `noncompliant` values for the `compliance_status` `input` key, but Intune has additional fields that need to be supported: "unknown", "conflict", "error", "ingraceperiod".

To solve this, we will update the documentation to state that the `compliance_status` key is utilized by both the Workspace One and Intune posture providers, and add support for the additional values that Intune provides.

`connection_id` is a different `input` field that is also shared by both workspace one and intune. We will update the documentation to reflect this as well.

### CODE REVIEW

**Comment by da-cf** [comment]
Closing due to potential implementation changes. May re-open.

**Comment by da-cf** [comment]
Re-opening as we will be moving forward with this implementation.

File: internal/sdkv2provider/resource_cloudflare_device_posture_rule_test.go [hunk 1]
```diff
@@ -289,6 +289,41 @@ func TestAccCloudflareDevicePostureRule_DiskEncryption_CheckDisks(t *testing.T)
 	})
 }
 
+func TestAccCloudflareDevicePostureRule_Intune(t *testing.T) {
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_device_posture_rule_test.go:292]: i'm getting an acceptance test failure here

```
=== RUN   TestAccCloudflareDevicePostureRule_Intune
    resource_cloudflare_device_posture_rule_test.go:303: Step 1/1 error: Error running apply: exit status 1
        
        Error: error creating Device Posture Rule for account "f037e56e89293a057740de681ac9abbe": Error: Invalid connection id (6117)
        
          with cloudflare_device_posture_rule.rbisfrrpde,
          on terraform_plugin_test.tf line 12, in resource "cloudflare_device_posture_rule" "rbisfrrpde":
          12: resource "cloudflare_device_posture_rule" "rbisfrrpde" {
        
--- FAIL: TestAccCloudflareDevicePostureRule_Intune (0.83s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	41.643s
FAIL
make: *** [testacc] Error 1
```

do we need other setup steps here? or is this something that requires manual setup and we shouldn't run all the time for the default account?

Comment (da-cf) [reply internal/sdkv2provider/resource_cloudflare_device_posture_rule_test.go:292]: The issue was my lack of understanding of the correct usage of connection_id. After making corrections, this test is now passing on my end:
```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareDevicePostureRule_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareDevicePostureRule_SerialNumber
--- PASS: TestAccCloudflareDevicePostureRule_SerialNumber (3.56s)
=== RUN   TestAccCloudflareDevicePostureRule_OsVersion
--- PASS: TestAccCloudflareDevicePostureRule_OsVersion (2.94s)
=== RUN   TestAccCloudflareDevicePostureRule_OsVersionExtra
--- PASS: TestAccCloudflareDevicePostureRule_OsVersionExtra (3.61s)
=== RUN   TestAccCloudflareDevicePostureRule_LinuxOsDistro
--- PASS: TestAccCloudflareDevicePostureRule_LinuxOsDistro (3.79s)
=== RUN   TestAccCloudflareDevicePostureRule_DomainJoined
--- PASS: TestAccCloudflareDevicePostureRule_DomainJoined (4.12s)
=== RUN   TestAccCloudflareDevicePostureRule_Firewall
--- PASS: TestAccCloudflareDevicePostureRule_Firewall (3.62s)
=== RUN   TestAccCloudflareDevicePostureRule_DiskEncryption_RequireAll
--- PASS: TestAccCloudflareDevicePostureRule_DiskEncryption_RequireAll (4.19s)
=== RUN   TestAccCloudflareDevicePostureRule_DiskEncryption_CheckDisks
--- PASS: TestAccCloudflareDevicePostureRule_DiskEncryption_CheckDisks (3.82s)
=== RUN   TestAccCloudflareDevicePostureRule_Intune
--- PASS: TestAccCloudflareDevicePostureRule_Intune (4.23s)
PASS
ok      github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider      34.437s
```

**Comment by jacobbednarz** [comment]
have skipped the new assertion given it requires an Intune license.

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareDevicePostureRule_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareDevicePostureRule_SerialNumber
--- PASS: TestAccCloudflareDevicePostureRule_SerialNumber (5.42s)
=== RUN   TestAccCloudflareDevicePostureRule_OsVersion
--- PASS: TestAccCloudflareDevicePostureRule_OsVersion (3.91s)
=== RUN   TestAccCloudflareDevicePostureRule_OsVersionExtra
--- PASS: TestAccCloudflareDevicePostureRule_OsVersionExtra (4.29s)
=== RUN   TestAccCloudflareDevicePostureRule_LinuxOsDistro
--- PASS: TestAccCloudflareDevicePostureRule_LinuxOsDistro (4.23s)
=== RUN   TestAccCloudflareDevicePostureRule_DomainJoined
--- PASS: TestAccCloudflareDevicePostureRule_DomainJoined (3.55s)
=== RUN   TestAccCloudflareDevicePostureRule_Firewall
--- PASS: TestAccCloudflareDevicePostureRule_Firewall (3.86s)
=== RUN   TestAccCloudflareDevicePostureRule_DiskEncryption_RequireAll
--- PASS: TestAccCloudflareDevicePostureRule_DiskEncryption_RequireAll (3.96s)
=== RUN   TestAccCloudflareDevicePostureRule_DiskEncryption_CheckDisks
--- PASS: TestAccCloudflareDevicePostureRule_DiskEncryption_CheckDisks (4.27s)
=== RUN   TestAccCloudflareDevicePostureRule_Intune
    provider_test.go:217: Skipping acceptance test for default account (f037e56e89293a057740de681ac9abbe). Assertion requires active Intune license.
--- SKIP: TestAccCloudflareDevicePostureRule_Intune (0.00s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	34.521s
```

**Pull request merged**